code/modules/almayer/weaponhits.dm

```diff
@@ -15,109 +13,47 @@
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), current_mob.client, SPAN_DANGER("You hear the Point Defense systems shooting down a missile!")), shotspacing SECONDS)
-
-		if(WEAPON_RAILGUN)
-			if(!weaponmiss)
-				for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
-					if(!is_mainship_level(current_mob.z))
-						continue
-					playsound_client(current_mob.client, 'sound/effects/bigboom3.ogg', 50)
-					playsound_client(current_mob.client, 'sound/effects/railgunhit.ogg', 50)
-					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound_client), current_mob.client, 'sound/effects/double_klaxon.ogg'), 2 SECONDS)
+				echo_list[ECHO_OBSTRUCTION] = -5000
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/railgun_miss.ogg', 5, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/laser_point_defence_success.ogg', 15, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+			sleep(2 SECONDS)
+		shipwide_ai_announcement("[hit] railgun projectile[hit > 1 ? "s" : ""] hit and [length(location) - hit] miss / intercepred", MAIN_AI_SYSTEM, 'sound/effects/alert.ogg')
+	else
+		for(var/turf/picked_atom in location)
+			if(prob(hitchance))
+				echo_list[ECHO_OBSTRUCTION] = -500
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(cell_explosion), picked_atom, 500, 10, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, cause_data))
+				shakeship(10, 10, TRUE, FALSE)
+				playsound(picked_atom, "bigboom", 100, 1, 200, echo_list)
+				playsound(picked_atom, 'sound/effects/metal_crash.ogg', 100, 1, 200, echo_list)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, "pry", 25, 1, 200, VOLUME_SFX, 0, null, 1, echo_list), 1 SECONDS)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/double_klaxon.ogg', 25, 1, 200, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+				hit++
 			else
-				for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
-					if(!is_mainship_level(current_mob.z))
-						continue
-					playsound_client (current_mob.client, 'sound/effects/railgun_miss.ogg', 60)
-					to_chat(current_mob.client, SPAN_DANGER("You hear railgun shots barely missing the hull!"))
-//REMOVE THIS WHEN WE USE THESE DEFS SOMEWHERE ELSE OR ELSE IT STRAIGHT UP WON'T WORK.
-#undef WEAPON_MISSILE
-#undef WEAPON_RAILGUN
+				echo_list[ECHO_OBSTRUCTION] = -5000
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/metal_shatter.ogg', 5, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(playsound), picked_atom, 'sound/effects/laser_point_defence_success.ogg', 15, 1, 100, VOLUME_SFX, 0, null, 1, echo_list), 2 SECONDS)
+			sleep(2 SECONDS)
```

Comment:
You do not need to sleep, just increase the callback timers for each volley and have the final announcement do the same.

Reply:
No you don't understand. Increased the timer callbacks so you don't require a sleep.

Reply:
Make for all proc callbacks... bruh, that wasting time

Reply:
Do you realize that this will stress the server more than the sleep() process?